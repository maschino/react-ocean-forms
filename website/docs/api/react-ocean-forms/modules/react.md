---
id: "react"
title: "react"
sidebar_label: "react"
---

Copyright (c) 2018-present, Umweltbundesamt GmbH

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.

## Index

### Type aliases

* [PropsOf](react.md#propsof)

## Type aliases

###  PropsOf

Æ¬ **PropsOf**: *T extends function ? P : T extends object ? P : never*

Acquire the props for a Component {T}
