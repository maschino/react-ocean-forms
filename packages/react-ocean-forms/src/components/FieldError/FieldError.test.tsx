import React from 'react';

import { render } from '@testing-library/react';

import { FieldError } from './FieldError';
import { Form } from '../Form';

describe('<FieldError />', () => {
  it('should do nothing if invalid is false', () => {
    const { asFragment } = render(
      <Form>
        <FieldError id="unitError" invalid={false} error={null} />
      </Form>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  const errorId = 'foo';
  const errorParams = { foo: 'bar' };

  it('should render an error', () => {
    const { getByText } = render(
      <Form>
        <FieldError
          id="unitError"
          invalid
          error={{
            message_id: errorId,
            params: errorParams,
          }}
        />
      </Form>
    );

    expect(getByText('foo')).toBeVisible();
  });

  it('should correctly use the stringFormatter', () => {
    const mockStringFormatter = jest.fn();

    render(
      <Form formatString={mockStringFormatter}>
        <FieldError
          id="unitError"
          invalid
          error={{
            message_id: errorId,
            params: errorParams,
          }}
        />
      </Form>
    );

    expect(mockStringFormatter).toHaveBeenCalledTimes(1);
    expect(mockStringFormatter).toHaveBeenCalledWith(errorId, errorParams);
  });

  it('should render multiple errors', () => {
    const { getByText } = render(
      <Form>
        <FieldError
          id="unitError"
          invalid
          error={[
            {
              message_id: errorId,
              params: errorParams,
            },
            { message_id: 'foo2', params: {} },
          ]}
        />
      </Form>
    );

    expect(getByText('foo')).toBeVisible();
    expect(getByText('foo2')).toBeVisible();
  });
});
