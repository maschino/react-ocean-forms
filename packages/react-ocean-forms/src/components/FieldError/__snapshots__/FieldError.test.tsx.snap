// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldError /> should do nothing if invalid is false 1`] = `
<DocumentFragment>
  <form
    class=""
  />
</DocumentFragment>
`;
